apps/dashboard/app/new-2/hooks/use-workspace-step.tsx

```diff
@@ -12,32 +17,102 @@ const workspaceSchema = z.object({
     .trim()
     .min(3, "Workspace name is required")
     .max(50, "Workspace name must be 50 characters or less"),
-  workspaceUrl: z
-    .string()
-    .min(3, "Workspace URL is required")
-    .regex(
-      /^[a-zA-Z0-9-_]+$/,
-      "URL handle can only contain letters, numbers, hyphens, and underscores",
-    ),
+  // workspaceUrl: z
+  //   .string()
+  //   .min(3, "Workspace URL is required")
+  //   .regex(
+  //     /^[a-zA-Z0-9-_]+$/,
+  //     "URL handle can only contain letters, numbers, hyphens, and underscores"
+  //   ),
 });
 
 type WorkspaceFormData = z.infer<typeof workspaceSchema>;
 
 export const useWorkspaceStep = (): OnboardingStep => {
-  const [isSlugGenerated, setIsSlugGenerated] = useState(false);
+  // const [isSlugGenerated, setIsSlugGenerated] = useState(false);
   const formRef = useRef<HTMLFormElement>(null);
+  const router = useRouter();
+  const searchParams = useSearchParams();
+  const [isPending, startTransition] = useTransition();
+  const workspaceIdRef = useRef<string | null>(null);
 
   const form = useForm<WorkspaceFormData>({
     resolver: zodResolver(workspaceSchema),
     defaultValues: {
-      workspaceName: "",
-      workspaceUrl: "",
+      workspaceName: searchParams?.get("workspaceName") || "",
     },
     mode: "onChange",
   });
 
-  const onSubmit = (data: WorkspaceFormData) => {
-    console.info("Workspace form submitted:", data);
+  const switchOrgMutation = trpc.user.switchOrg.useMutation({
+    onSuccess: async (sessionData) => {
+      if (!sessionData.expiresAt) {
+        console.error("Missing session data: ", sessionData);
+        toast.error(`Failed to switch organizations: ${sessionData.error}`);
+        return;
+      }
+
+      await setCookie({
```

Comment:
I'd appreciate if @mcstepp or @perkinsjr can help review this, cause I'm not too deep into the frontend auth inner workings

Reply:
Btw, this is just mimicking old create -> https://github.com/unkeyed/unkey/blob/main/apps/dashboard/app/new/create-workspace.tsx. tRPC logic is the same